src/basic/hashmap.c

```diff
@@ -1794,10 +1794,10 @@ int set_consume(Set *s, void *value) {
         return r;
 }
 
-int _hashmap_put_strdup(Hashmap **h, const char *k, const char *v  HASHMAP_DEBUG_PARAMS) {
+int _hashmap_put_strdup_full(Hashmap **h, bool use_string_hash, const char *k, const char *v  HASHMAP_DEBUG_PARAMS) {
         int r;
 
-        r = _hashmap_ensure_allocated(h, &string_hash_ops_free_free  HASHMAP_DEBUG_PASS_ARGS);
+        r = _hashmap_ensure_allocated(h, use_string_hash ? &string_hash_ops_free_free : &trivial_hash_ops_free_free  HASHMAP_DEBUG_PASS_ARGS);
```

Comment:
hmm, instead of taking a bool argument here and converting that in one of two hash ops structs, maybe just take a hash opt ptr right-away as argument?

That way it woud be more generic I figure, and probably even more descriptive since a standalone "false" or "true" as one of many params in a function call is always a bit opaque, but a parameter such as `&trivial_hash_ops_free_free` quite a bit more explanatory

Reply:
Thank you for the advice. Updated.